compiler/parser/RdrHsSyn.hs

```diff
@@ -2881,11 +2884,11 @@ mkTemplateDecls header fields decls = do
   ci@(conName, _, _) <- splitCon [fields, rdrNameToType vtTemplateName]
   -- Create choice data types except for Archive, which has a single definition across templates
   choiceDataDecls <- concat <$> traverse mkChoiceDataDecls vtChoices
-  let templateName = rdrNameToString vtTemplateName
-      tInstanceClass = unLoc $ mkUnqualClass $ mkInstanceClassName templateName
+  let templateName = occNameString . rdrNameOcc <$> vtTemplateName
+      tInstanceClass = unLoc $ mkUnqualClass $ mkInstanceClassName <$> templateName
       templateDataDecl = mkTemplateDataDecl (combineLocs vtTemplateName fields) vtTemplateName vtTypeVars ci
-      choicesWithArchive = mkArchiveChoice templateName : vtChoices
-      templateInstClassDecl = mkTemplateInstanceClassDecl (getLoc vtTemplateName) conName vt{vtChoices = choicesWithArchive}
+      choicesWithArchive = mkArchiveChoice (unLoc templateName) : vtChoices
```

Comment:
Why strip the location information? If the archive choices points to the template that's definitely better than pointing nowhere.

Reply:
True. This change preserves lack of location for this case, but there is definitely more work to include locations for archive and the other template methods.